PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -1519,6 +1519,106 @@ HFONT createFont(const TCHAR* fontName, int fontSize, bool isBold, HWND hDestPar
 	return newFont;
 }
 
+bool isRawWin32FileName(const generic_string& fileName)
+{
+	// TODO:
+	// ?! how to handle similar NT Object Manager path style prefix case \??\...
+	// (the \??\ prefix instructs the NT Object Manager to search in the caller's local device directory for an allias...)
+
+	// the following covers the \\?\... raw Win32-filenames or the \\?\UNC\... UNC equivalents
+	// and also its *nix like forward slash equivalents
+	return (fileName.starts_with(TEXT("\\\\?\\")) || fileName.starts_with(TEXT("//?/")));
+}
+
+bool isRawWin32FileName(const TCHAR* szFileName)
+{
+	const generic_string fileName = szFileName;
+	return isRawWin32FileName(fileName);
+}
+
+const generic_string reservedWinOSNamesList[] = {
+	TEXT("CON"), TEXT("PRN"), TEXT("AUX"), TEXT("NUL"),
+	TEXT("COM1"), TEXT("COM2"), TEXT("COM3"), TEXT("COM4"), TEXT("COM5"), TEXT("COM6"), TEXT("COM7"), TEXT("COM8"), TEXT("COM9"),
+	TEXT("LPT1"), TEXT("LPT2"), TEXT("LPT3"), TEXT("LPT4"), TEXT("LPT5"), TEXT("LPT6"), TEXT("LPT7"), TEXT("LPT8"), TEXT("LPT9")
+};
+const std::vector<generic_string> rwnlVector(reservedWinOSNamesList, reservedWinOSNamesList + sizeof(reservedWinOSNamesList) / sizeof(reservedWinOSNamesList[0]));

```

Comment:
And the variable `reservedWin32DeviceNamespaceList` can be declared as a local variable in  `isUnsupportedFileName` function, instead of being a global variable.